app/models/service/resource_linking.rb

```diff
@@ -0,0 +1,37 @@
+module Service::ResourceLinking
+  extend ActiveSupport::Concern
+
+  def add_provider_vms(options)
+    # required keys in options hash:
+    #   :target_id, the id of the provider to refresh
+    #   :uid_ems_array, an array of VMs' uid_ems
```

Comment:
Why are we taking these as a hash and not just normal parameters?  Seems we are doing the `raise` in `normalize_inputs!` for `target_id` when it should just be a normal argument (or keyword args).  I realize you are merging the options but the existing `normalize_inputs!` method could return the hash that gets merged.

Maybe something like: `add_provider(target_id:, uid_ems_array:)` but depends on what the API is passing.

@jntullo Are you passing the provider object or just the ID from the API?

Reply:
@gmcculloug I was passing in the actual provider object with the `:uid_ems_array` but see that it has changed

Reply:
If we need the `zone` from the provider we'll need to load the object.  Let's understand that first before we change what you are passing.